trunk/NDHMS/Routing/module_RT.F

```diff
@@ -796,6 +798,36 @@ subroutine LandRT_ini(did)
 
        end select
 
+     else if (rt_domain(did)%q_type(lake_index) == 2) then
+
+       ! Allocate a single persistence_levelpool_hybrid reservoir
+       allocate(persistence_levelpool_hybrid_struct :: rt_domain(did)%reservoirs(lake_index)%ptr)
+
+       ! Dynamically inspects for persistence_levelpool_hybrid type reservoir
+       select type (hybrid => rt_domain(did)%reservoirs(lake_index)%ptr)
+          type is (persistence_levelpool_hybrid_struct)
```

Comment:
This is sort of funny given line 804 is right there. Can we move line 804 up to combine with line 777 allocate in the q_type and do the allocate in `if elseif` conditionals. Then move the `select` out of the conditional? It's just redundant to have that select inside the ifelse. Maybe there's some reason this cant be done currently? It looks perty straight forward. It will really make tight the meaning of `q_type`.

Reply:
@jmccreight

Reply:
Fortran is sort of stupid about pointers. What is happening here is that rt_domain(did)%reservoirs(lake_index)%ptr is a pointer to the base class of reservoirs. The allocation at line 777 allocates a persistence hybrid reservoir object on that pointer but it is still only a base class pointer. It can be used to call virtual methods but init can not be virtual so a pointer to the actual class is required.

In Fortran this can only be achieved using a select type statement. In C++ you would just use dynamic_cast<levelpool_persistance_hybrid*>(reservoir_base_ptr).

Reply:
I changed this in be5da22: this is not functionally different, but it is easier to read.

Reply:
You can see it here. 
https://github.com/NCAR/wrf_hydro_nwm_public/pull/259/files?file-filters%5B%5D=.F#diff-d9becf4dff01449d59389925795848caR774

Reply:
I see you used the select type to replace some of the original if/else if logic. Yes this works fine, the reason it was coded as is was to minimize changes to the code outside of the reservoir module. I agree doing it this is cleaner.